src/shared/lib/redux/types.ts

```diff
@@ -0,0 +1,12 @@
+import { Action, ThunkDispatch } from '@reduxjs/toolkit';
+import { useDispatch } from 'react-redux';
+
+import { reduxStore } from '@app/app/providers';
```

Comment:
This import is prohibited since "app" is the top-level layer. I'd suggest you move these types to ReduxProvider.
You can also declare there RootState and AppDispatch as global types